backend/src/TodosDatabase.ts

```diff
@@ -1,60 +1,77 @@
 import { TodoDto } from './dto/todoDto';
 import { ConnectionService } from './app.connectionService';
+import { Injectable } from '@nestjs/common';
+import { ConflictException } from '@nestjs/common';
+import { TodoDatabaseRow } from './dto/todoDto';
 
+@Injectable()
 export default class TodosDatabase {
-  private static connectionService = new ConnectionService();
+  private connectionService = new ConnectionService();
 
-  static async createTodo(todo: TodoDto) {
-    const newTodoOrder = await this.getNextTodoMaxOrder();
-    console.log(newTodoOrder);
+  async upsertTodo(todo: TodoDto, todoOrder?: number) {
     const insertTodoQuery = `
     INSERT INTO "todosTable" (id,title,date,status,todosorder)
     VALUES ($1,$2,$3,$4,$5)
+    ON CONFLICT (id)
+    DO UPDATE SET title = EXCLUDED.title,date = EXCLUDED.date,status = EXCLUDED.status
     RETURNING *;`;
-    const todoValues = [
-      todo.id,
-      todo.title,
-      todo.date,
-      todo.status,
-      newTodoOrder,
-    ];
-    return this.connectionService.query(insertTodoQuery, todoValues);
+    const todoValues = [todo.id, todo.title, todo.date, todo.status, todoOrder];
+    return this.connectionService
+      .query(insertTodoQuery, todoValues)
+      .then((query) => {
+        const todosAffected = query.rows;
+        if (todosAffected.length !== 1) {
+          throw new ConflictException('conflict happend', {
+            cause: new Error(),
+            description: 'request yeilded more than one result',
+          });
+        }
+        return todosAffected[0] as TodoDatabaseRow;
+      });
   }
 
-  private static async getNextTodoMaxOrder(): Promise<number> {
+  public async getCurrentTodoMaxOrder() {
```

Comment:
i don't think that you need the public

Reply:
I calculate the next index at the service class using this function

Reply:
try to remove, i don't think it is needed, public is the default modifier AFAIR